File: test-fixtures/languages/erlang/micro/user_service.erl
-module(user_service).
-export([get_user/1, update_user/2]).

get_user(Token) ->
    auth_service:verify_token(Token).

update_user(Token, Action) ->
    UserId = auth_service:verify_token(Token),
    case UserId of
        undefined -> undefined;
        _ ->
            auth_service:run_admin_command(UserId, Action),
            UserId
    end.

File: test-fixtures/languages/rust/micro/user_service.rs
use crate::micro::auth_service::{verify_token, run_admin_command};

pub struct UserInfo {
    pub id: String,
    pub token: String,
}

pub fn get_user(conn: &rusqlite::Connection, token: &str) -> Option<UserInfo> {
    let user_id = verify_token(conn, token)?;
    Some(UserInfo {
        id: user_id,
        token: token.to_string(),
    })
}

pub fn update_user(conn: &rusqlite::Connection, token: &str, action: &str) -> Option<String> {
    let user_id = verify_token(conn, token)?;
    run_admin_command(&user_id, action);
    Some(user_id)
}
